platform/linux-generic/arch/aarch64/cpu_flags.c

```diff
@@ -395,7 +1122,118 @@ static void _odp_sys_info_print_acle_flags(void)
 	ODP_PRINT("\n");
 }
 
+static int check_hwcap_duplicates(unsigned int hwcap_field)
+{
+	int ret = 0;
+
+	/* FP and AdvSIMD fields of the AArch64 Processor
+	 * Feature Register 0 must have the same value and are
+	 * defined by the same feature flag. Print the flag
+	 * only once. */
+#ifdef HWCAP_ASIMD
+		if (hwcap_field == HWCAP_ASIMD)
+			ret = 1;
+#endif
+#ifdef HWCAP_ASIMDHP
+		if (hwcap_field == HWCAP_ASIMDHP)
+			ret = 1;
+#endif
+
+	return ret;
+}
+
+static void _odp_sys_info_print_hwcap_flags(void)
+{
+	unsigned long hwcaps, hwcaps2;
+	unsigned int size, size2;
+
+	/* Check if the hwcap_flags[] is an empty array. If true, the function
+	 * will return. This is unlikely behaviour. */
+	size = sizeof(hwcap_flags) / sizeof(hwcap_feat_flag_t);
+	if (size == 0)
+		return;
+
+	/* AT_HWCAP flag is supported. Print the flags supported by hardware. */
+	ODP_PRINT("ARM FEATURES SUPPORTED BY HARDWARE:\n");
+
+	hwcaps = getauxval(AT_HWCAP);
+	for (unsigned int i = 0; i < size; i++) {
+		if (hwcap_flags[i].valid) {
+			if (check_hwcap_duplicates(hwcap_flags[i].hwcap_field)) {
+				hwcaps = hwcaps >> 1;
+				continue;
+			}
+
+			if (hwcaps & 0x01)
+				ODP_PRINT("%s ", hwcap_flags[i].feat_flag);
+			hwcaps = hwcaps >> 1;
+		}
+	}
+
+	/* Check if the hwcap2_flags[] is an empty array. If true, the function
+	 * will print out the unsupported flags for AT_HWCAP and exit. Otherwise,
+	 * print supported flags for AT_HWCAP2.
+	 * CentOS/RedHat systems do not support AT_HWCAP2 flag. */
```

Comment:
The second sentence is deprecated and I'd assume that `AT_HWCAP2` is supported by newer RedHat releases. Maybe it's easiest to remove the whole comment.

Reply:
I will remove it. Thanks!